proto/src/determined/trial/v1/trial.proto

```diff
@@ -148,3 +151,78 @@ message TrialEarlyExit {
   // The reason for the exit.
   ExitedReason reason = 1;
 }
+
+// The training metrics for some duration of training.
+message TrainingMetrics {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: {
+      required: [ "trial_id", "trial_run_id", "batches_trained", "metrics" ]
+    }
+  };
+  // The trial associated with these metrics.
+  int32 trial_id = 1;
+  // The trial run associated with these metrics.
+  int32 trial_run_id = 2;
+  // The number of batches trained on when these metrics were reported.
+  int32 batches_trained = 3;
+  // The number of batches trained on when these metrics were reported.
+  int32 records_trained = 4;
+  // The metrics for this bit of training (reduced over the reporting period).
+  google.protobuf.Struct metrics = 5;
+  // The batch metrics for this bit of training.
+  repeated google.protobuf.Struct batch_metrics = 6;
+}
+
+// The validation metrics for some duration of training.
+message ValidationMetrics {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: {
+      required: [ "trial_id", "trial_run_id", "batches_trained", "metrics" ]
+    }
+  };
+  // The trial associated with these metrics.
+  int32 trial_id = 1;
+  // The trial run associated with these metrics.
+  int32 trial_run_id = 2;
+  // The number of batches trained on when these metrics were reported.
+  int32 batches_trained = 3;
+  // The number of batches trained on when these metrics were reported.
+  int32 records_trained = 4;
+  // The metrics.
+  google.protobuf.Struct metrics = 5;
+}
+
+// The checkpoint metadata for some trial checkpoint.
+message CheckpointMetadata {
+  // XXX: Clean up required on all these.
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: {
+      required: [
+        "trial_id",
```

Comment:
is there a plan for how we will migrate checkpoints away from depending on training-specific values, like trial_id?

Reply:
i think we will remove the "required" here which is a backwards compatible change. other than that, i think we can figure it out when we get there.

Reply:
wait, removing required in a response though isn't backwards compatible.. hmm. idk, i think we're gonna need a breaking change for that tbh, either that or we make v2 apis when we promote this out of internal